grid_map_loader/src/GridMapLoader.cpp

```diff
@@ -30,27 +38,31 @@ GridMapLoader::~GridMapLoader()
 
 bool GridMapLoader::readParameters()
 {
-  nodeHandle_.param("bag_topic", bagTopic_, std::string("/grid_map"));
-  nodeHandle_.param("publish_topic", publishTopic_, bagTopic_);
-  nodeHandle_.param("file_path", filePath_, std::string());
   double durationInSec;
-  nodeHandle_.param("duration", durationInSec, 5.0);
-  duration_.fromSec(durationInSec);
+  this->declare_parameter("bag_topic", std::string("/grid_map"));
+  this->declare_parameter("publish_topic", std::string("/grid_map"));
+  this->declare_parameter("file_path", std::string());
+  this->declare_parameter("duration", rclcpp::ParameterValue(5.0));
+
+  this->get_parameter("bag_topic", bagTopic_);
+  this->get_parameter("publish_topic", publishTopic_);
+  this->get_parameter("file_path", filePath_);
+  this->get_parameter("duration", durationInSec);
+
+  duration_ = rclcpp::Duration::from_seconds(durationInSec);
   return true;
 }
 
 bool GridMapLoader::load()
 {
-  ROS_INFO_STREAM("Loading grid map from path " << filePath_ << ".");
-  return GridMapRosConverter::loadFromBag(filePath_, bagTopic_, map_);
+  RCLCPP_INFO_STREAM(this->get_logger(), "Loading grid map from path " << filePath_ << ".");
+  return grid_map::GridMapRosConverter::loadFromBag(filePath_, bagTopic_, map_);
 }
 
 void GridMapLoader::publish()
 {
-  grid_map_msgs::GridMap message;
+  grid_map_msgs::msg::GridMap message;
   grid_map::GridMapRosConverter::toMessage(map_, message);
-  publisher_.publish(message);
-  duration_.sleep();
+  publisher_->publish(message);
```

Comment:
Removed sleep? 

Also the publish should be a unique_ptr if possible. Can we change the `toMessage` signature?

Reply:
publish updated!

Changing `toMessage` to take in a unique_ptr as in `toMessage(..., unique_ptr<msg> message)`?

Reply:
Uh that's not good `publisher_->publish(std::make_unique<grid_map_msgs::msg::GridMap>(message));` that's a big copy. you should change the signature of toMessage to return a unique pointer. For instance something really readable would be:

publisher_->publish(std::move(grid_map::GridMapRosConverter::toMessage(map_)));

Reply:
Got it.